### ISSUE TITLE
Please address how to setup NetworkManager-Strongswan plugin With DNS upstream together

### ISSUE THREAD

**Issue by leeee-me** [jawj/IKEv2-setup#98]
I tried to configure it correctly because my client Ubuntu used NetworkManager for everything. By doing so, it will combine together the correct ip routing and DNS upstream integration:
[vpn]
address=$VPNHOSTADDR
encap=no
esp=aes256gcm16
ike=aes256gcm16-prfsha384-ecp521
ipcomp=no
method=eap
password-flags=0
proposal=yes
user=$VPNUSERNAME
virtual=yes
service-type=org.freedesktop.NetworkManager.strongswan

[vpn-secrets]
password=$VPNPASSWORD

However, this NetworkManager stuff (and your vpn-ubunut-client.sh) cannot correctly combine the VPN DNS from VPN host.
In my NetworkManager conf, dns is controlled by associated D-Bus dnsmasq worker.

Do need to integrated with several DNS resolver such as systemd-resolved, dnsmasq, resolvconf, etc

Thanks

**Comment by jawj** [comment]
Afraid I don't know the answer here, but if you find out do let me know and I'll add it to the script or README.

**Comment by leeee-me** [comment]
I indeed found a way to configure correctly network-manager-strongswan plugin. However, there are two things which can be seen as bugs for this specific configuration which needs to be solved first on my Ubuntu 18.04.03 LTS before it works:
1). I must use the up-to-date 5.8.1 debian package and this can fix the VPN DNS integration issue which those DNS info broadcasted by remote VPN server. In this case, I downloaded debian sources, and rebuilt a whole new set of strongswan families: 
libstrongswan 5.8.1-1ubuntu1
libstrongswan-extra-plugins 5.8.1-1ubuntu1
libstrongswan-standard-plugins 5.8.1-1ubuntu1
network-manager-strongswan 1.4.4-2
strongswan 5.8.1-1ubuntu1
strongswan-charon 5.8.1-1ubuntu1
strongswan-libcharon 5.8.1-1ubuntu1
strongswan-nm 5.8.1-1ubuntu1
strongswan-pki 5.8.1-1ubuntu1
strongswan-scepclient 5.8.1-1ubuntu1
strongswan-starter 5.8.1-1ubuntu1
strongswan-swanctl 5.8.1-1ubuntu1
strongswan-tnc-base 5.8.1-1ubuntu1
strongswan-tnc-client 5.8.1-1ubuntu1
strongswan-tnc-ifmap 5.8.1-1ubuntu1
strongswan-tnc-pdp 5.8.1-1ubuntu1
strongswan-tnc-server 5.8.1-1ubuntu1

2). This IKEv2 VPN did not work for a Linux client behind a WWAN modem ISP (e.g., a Linux with USB 4G/LTE modem, connected to the Internet by ISP). I am sure this is not caused by ISP because I have MacOS and iOS device which have 4G/LTE modem connected to the Internet by the same ISP. The charon/ipsec encryption algorithm negotiation step will always failed, which never have a good session initialization. Very strangely, this issue will not happen on a Linux client which is behind a normal NAT of WiFi or ADSL Access Point (via the same strongswan plugin and netowkr-manager). I already reported this issue at Ubuntu's launchpad. I supposed this is an issue of network-manager which is incapable to handle wwan and IKEv2/strogswan together.
